### ISSUE TITLE
Updater does not decode Content-Encoding before writing files

### ISSUE THREAD

**Issue by Geemer** [FlashpointProject/FlashpointUltimateUpdater#10]
The updater is putting garbage data in a bunch of files and breaking the loading of games/animations. 

Flashpoint will start complaining the preferences are bad and offer to load default preferences, then it'll be full of placeholder strings.

The files I've seen affected include changelog.txt and preferences.json, as well as a lot (all?) of the language packs. The copies on the server seem to be fine, but replacing the preferences and en.json files still doesn't result in a functional Flashpoint. 
It seems like some important core files are also affected. 

I've attached a screenshot of the af-ZA.json language file that the updater made side by side to the one on the server. As well as a picture of how it loads when en.json is still broken.

![Clipboard01](https://user-images.githubusercontent.com/6912397/200190158-dd6dc37f-8e5f-4be4-b569-06f3b514915e.png)
![Clipboard02](https://user-images.githubusercontent.com/6912397/200190160-ec666ffe-e90d-4107-96be-c88975fcbd63.png)

**Comment by XXLuigiMario** [comment]
Yeah.. that's bad. I wonder if it has something to do with the server sending the files with an unsupported encoding or compression method? I recall having a similar issue in the past, regarding brotli compression. Could you upload an affected file?

**Comment by Geemer** [comment]
Here's the af-ZA.json I took a screenshot of earlier. Should've thought to just upload it instead.

Don't mind the double extension, I had to tack on the .txt for GitHub to let me upload it.

[af-ZA.json.txt](https://github.com/FlashpointProject/FlashpointUltimateUpdater/files/9946631/af-ZA.json.txt)

**Comment by XXLuigiMario** [comment]
Yeah, just what I thought. It's not garbage, it's compressed. The would be why is the requests library is not decompressing it
> joel@blau:~/Downloads$ file af-ZA.json.txt 
af-ZA.json.txt: gzip compressed data, original size modulo 2^32 24044

**Comment by Geemer** [comment]
Sorry that it's been a while, finally had some time to mess with python to try the change you suggested. 

I needed to add `import functools` to the list of imports at the top of the script as well to get it to work, but it seems to have fixed the issue. 

I guess my path forward is reverting back to version 10.1 and then redoing the update to 11 to fix all the files that got stuck as gzip.

**Comment by XXLuigiMario** [comment]
Actually, if you just rerun it pointing to version 11 it should deal with it, no need to go back to version 10.1.

**Comment by Geemer** [comment]
Thanks, unfortunately your message came in after I had already decided to take a slightly different course of action: delete everything but the Data and Legacy folders and tell it to re-download version 11.

However, I still had problems starting anything. Turns out some of the files under Data were still in gzip purgatory. 
Specifically the .json and .js files. So, credits.json, and in Extensions all the package.json and ext.js files.

Fixing those manually finally got my install working again.


I'm not sure if the updater going over the files again without downloading them actually fixes the gzipped files since it seems to only look at filenames, but I'm also not gonna mess with it for testing since just the indexing step at the start already takes hours.
It might also be worth putting out an update of the Windows binary for those who don't have Python laying around.
